feat: BROS-189: Add default open state to Collapse

nick-skriabin opened the pull request:
This pull request introduces a new `open` property to the `Collapse` component, enabling the default collapsed state to be set for individual panels. The changes span documentation updates, model adjustments, and rendering logic enhancements.

### Documentation Updates:
* Added the `open` property to the `Collapse` component's documentation, specifying its type as `boolean` and default value as `false`. This property sets the default collapsed state. (`docs/source/includes/tags/collapse.md`, [docs/source/includes/tags/collapse.mdR7](diffhunk://#diff-cb84d06bc6a0f2e0a5c7fd93604b818c8a6555730d20ef9f54b7ad7d5f22dcf0R7))

### Model Enhancements:
* Updated the `PanelModel` and `Model` in `Collapse.jsx` to include the `open` property as an optional boolean with a default value of `false`. (`web/libs/editor/src/tags/visual/Collapse.jsx`, [[1]](diffhunk://#diff-03936ec1b8164a425355cb4eca722e349b150e6ce802c137e5c0c1858beb3b31R36-R37) [[2]](diffhunk://#diff-03936ec1b8164a425355cb4eca722e349b150e6ce802c137e5c0c1858beb3b31R107)

### Rendering Logic Improvements:
* Modified the `HtxCollapse` component to calculate `defaultActiveKeys` based on the `open` property of child panels, ensuring the default collapsed state is respected during rendering. (`web/libs/editor/src/tags/visual/Collapse.jsx`, [web/libs/editor/src/tags/visual/Collapse.jsxR129-R140](diffhunk://#diff-03936ec1b8164a425355cb4eca722e349b150e6ce802c137e5c0c1858beb3b31R129-R140))

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Add default open state to collapse

robot-ci-heartex committed:
ci: Build tag docs

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16348331451

nick-skriabin commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16349322092)

nick-skriabin committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16349322092

nick-skriabin committed:
Change default open state to true

robot-ci-heartex committed:
ci: Build tag docs

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16349479770

nick-skriabin committed:
All panels closed by default

nick-skriabin committed:
Cleanup

robot-ci-heartex committed:
ci: Build tag docs

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16350325029

nick-skriabin committed:
Use collapse-level open state

nick-skriabin committed:
Proper global open handling

nick-skriabin committed:
Cleanup

hlomzik commented on web/libs/editor/src/tags/visual/Collapse.jsx:
```diff
@@ -32,6 +33,8 @@ const PanelModel = types
     _value: types.optional(types.string, ""),
     value: types.optional(types.string, ""),
 
+    open: types.optional(types.boolean, false),
```
This param should be optional and nullable then to allow global `open` to work

nick-skriabin committed:
Collapse tests

hlomzik commented on web/libs/editor/src/tags/visual/Collapse.jsx:
```diff
@@ -121,16 +126,33 @@ const CollapseModel = types.compose("CollapseModel", AnnotationMixin, Model, Pro
+      console.log(index, c.value, c.open, c.open ?? item.open);
+      if (isDefined(c.open)) return c.open;
+      return item.open;
+    })
```
```suggestion
    .filter((panel) => panel.open ?? item.open)
```

nick-skriabin committed:
Update web/libs/editor/src/tags/visual/Collapse.jsx

Co-authored-by: Andrew <hlomzik@gmail.com>

nick-skriabin commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16418505902)

nick-skriabin committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16418505902

nick-skriabin committed:
Merge branch 'develop' into fb-bros-189/collapse-default-state

nick-skriabin commented:
/fm merge
> Command | Description
> --- | ---
> /fm sync <Optional branch override> | Sync upstream prs and merge with pull request base
